sbncode/LArG4/SimpleMerge_module.cc

```diff
@@ -0,0 +1,103 @@
+////////////////////////////////////////////////////////////////////////
+// Class:       SimpleMerge
+// Plugin Type: producer (Unknown Unknown)
+// File:        producer_module.cc
+//
+// Generated at Tue Dec 19 08:26:07 2023 by Brinden Carlson using cetskelgen
+// from  version .
+//
+// Producer module for merging multiple input sources into a single output
+// without trackID offsets or other complications from merge sim sources.
+
+////////////////////////////////////////////////////////////////////////
+
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Principal/Run.h"
+#include "art/Framework/Principal/SubRun.h"
+#include "canvas/Utilities/InputTag.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+#include <memory>
+//#include <utility>
+
+#include "nusimdata/SimulationBase/MCParticle.h"
+#include "larcorealg/CoreUtils/enumerate.h"
+
+namespace sbn {
+  class SimpleMerge;
+}
+
+
+class sbn::SimpleMerge : public art::EDProducer {
+public:
+  explicit SimpleMerge(fhicl::ParameterSet const& p);
+  // The compiler-generated destructor is fine for non-base
+  // classes without bare pointers or other resource use.
+
+  // Plugins should not be copied or assigned.
+  SimpleMerge(SimpleMerge const&) = delete;
+  SimpleMerge(SimpleMerge&&) = delete;
+  SimpleMerge& operator=(SimpleMerge const&) = delete;
+  SimpleMerge& operator=(SimpleMerge&&) = delete;
+
+  // Required functions.
+  void produce(art::Event& e) override;
+
+private:
+
+  // Declare member data here.
+  std::vector<art::InputTag> const fInputSourcesLabels;
+  bool const fFillMCParticles;
+  std::vector<int> fTrackIDOffsets;
+  bool const fResetMotherID;
+
+};
+
+
+sbn::SimpleMerge::SimpleMerge(fhicl::ParameterSet const& p)
+  : EDProducer{p}
+  , fInputSourcesLabels(p.get<std::vector<art::InputTag>>("InputSourcesLabels"))
+  , fFillMCParticles(p.get<bool>("FillMCParticles", false))
+  , fTrackIDOffsets(p.get<std::vector<int>>("TrackIDOffsets", std::vector<int>(0)))
```

Comment:
I suppose an empty vector is intended here. Try one of the following:
```suggestion
  , fTrackIDOffsets(p.get<std::vector<int>>("TrackIDOffsets", {}))
```
or
```suggestion
  , fTrackIDOffsets(p.get("TrackIDOffsets", std::vector<int>{}))
```
or
```suggestion
  , fTrackIDOffsets(p.get<std::vector<int>>("TrackIDOffsets", std::vector<int>{}))
```
I _recommend_ avoiding `std::vector<int>(0)` because it looks ambiguous, whether you are initialising a vector with `0` elements or if you are initialising a vector with one element with value `0`.

Reply:
The intention is to have a vector with just 0 in there. Now that I look [here](https://github.com/SBNSoftware/sbncode/blob/3699a40018678bc7980ea111599b1a7655818660/sbncode/LArG4/SimpleMerge_module.cc#L87-L92) though, I see that if the mother ID is already 0, there's no point in setting it if the offset is set to 0. I suppose leaving it as a blank vector will be a good default!